internal/storage/image.go

```diff
@@ -398,6 +410,87 @@ func (svc *imageService) ImageStatus(systemContext *types.SystemContext, nameOrI
 	return &result, nil
 }
 
+// nolint:staticcheck
+func (svc *imageService) IsRunningImageAllowed(ctx context.Context, systemContext *types.SystemContext, configImage, userSpecifiedImage string) error {
+	if systemContext.SignaturePolicyPath == "" {
+		log.Infof(ctx, "No policy applied, skipping check if image is allowed to run")
+		return nil
+	} else if userSpecifiedImage == "" {
```

Comment:
nit: it's my preference to not have the else if we return in the previous branch, but if this is better for you :shrug:

Reply:
This does not exist any more.